File: 2021W37/B0neh3ad/pgr_12977.cpp
/*
    소수 만들기
    https://programmers.co.kr/learn/courses/30/lessons/12977
*/
#include <vector>
#include <iostream>
using namespace std;

int solution(vector<int> nums) {
    int answer = 0;
    // set primes
    int primes[3001] = {0, };
    for(int i=2; i*i<=3000; ++i)
        for(int j=i*i; j<=3000; j+=i)
            primes[j] = 1;
    // get answer
    for(int i=0; i<nums.size()-2; ++i)
        for(int j=i+1; j<nums.size()-1; ++j)
            for(int k=j+1; k<nums.size(); ++k)
                if(!primes[nums[i]+nums[j]+nums[k]])
                    ++answer;
    return answer;
}

File: 2022W26/brainer/pgr_12977.py
# 소수 만들기
# https://programmers.co.kr/learn/courses/30/lessons/12977

def is_prime(num):
    for i in range(2, num):
        if num % i == 0 and num != i:
            return False
    return True

def get_prim_num(nums):
    prim_list = []
    num_len = len(nums)
    
    for i in range(num_len):
        for j in range(i+1, num_len):
            for k in range(j+1, num_len):
                sum_num = nums[i] + nums[j] + nums[k]
                if is_prime(sum_num): 
                    prim_list.append(sum_num)
    return prim_list

def solution(nums):
    return len(get_prim_num(nums))
